packages/core-transactions/src/handlers/ipfs.ts

```diff
@@ -44,8 +45,9 @@ export class IpfsTransactionHandler extends TransactionHandler {
         wallet: State.IWallet,
         walletManager: State.IWalletManager,
     ): Promise<void> {
-        // TODO implement unique ipfs hash on blockchain (not just on wallet)
-        if (wallet.hasAttribute("ipfs") && wallet.getAttribute("ipfs.hashes")[transaction.data.asset.ipfs]) {
+        const ipfsHashes = walletManager.getIndex("ipfs").keys();
```

Comment:
You can use `walletManager.getIndex("ipfs").has(value)` then the lookup is O(1).

Reply:
Thanks!